tools/autograd/gen_variable_type.py

```diff
@@ -519,13 +519,14 @@ def find_args_with_derivatives(differentiable_inputs):
             if differentiable:
                 differentiable_outputs.append(output)
     elif uses_single_grad(func):
-        differentiable_outputs = candidate_differentiable_outputs[:1]
+        candidate_differentiable_outputs = candidate_differentiable_outputs[:1]
+        differentiable_outputs = candidate_differentiable_outputs
     else:
         differentiable_outputs = candidate_differentiable_outputs
 
     requires_derivative = (
         base_name not in DONT_REQUIRE_DERIVATIVE and name not in DONT_REQUIRE_DERIVATIVE and
-        len(differentiable_inputs) > 0 and len(differentiable_outputs) > 0 and
+        len(differentiable_inputs) > 0 and len(candidate_differentiable_outputs) > 0 and
```

Comment:
Why did this change? The code before made sense (if number of differentiable outputs is > 0, then requires_derivative is True)